src/configure/utilities/InputUxDescriptor.ts

```diff
@@ -0,0 +1,130 @@
+import { InputBoxOptions, QuickPickItem } from 'vscode';
+import { IAzureQuickPickOptions } from 'vscode-azureextensionui';
+import { telemetryHelper } from '../helper/telemetryHelper';
+import { ExtendedInputDescriptor, InputMode } from "../model/Contracts";
+import { AzureSession, ControlType, extensionVariables } from '../model/models';
+import { Messages } from '../resources/messages';
+import { TelemetryKeys } from '../resources/telemetryKeys';
+import { DataSourceExpression } from './DataSourceExpression';
+
+export class InputUxDescriptor {
```

Comment:
Would it be better to name it as `InputControl`?

Reply:
yes that looks better, renamed it.